### ISSUE TITLE
Fatal Errors in Logs after Upgrade from 10.1 to 10.2

### ISSUE THREAD

**Issue by commodis** [checkstyle/checkstyle#11755]
This report is not about a check and its pure `jar` execution in particular, but I assume some form of regression.

Running the following command

```bash
mvn --activate-profiles checkstyle checkstyle:check
```

with the maven profile

```xml
<profile>
    <id>checkstyle</id>

    <build>
        <pluginManagement>
            <plugins>
                <plugin>
                    <groupId>org.apache.maven.plugins</groupId>
                    <artifactId>maven-checkstyle-plugin</artifactId>
                    <version>3.1.2</version>

                    <configuration>
                        <configLocation>.checkstyle/config.xml</configLocation>
                        <includeTestSourceDirectory>true</includeTestSourceDirectory>
                    </configuration>

                    <dependencies>
                        <dependency>
                            <groupId>com.puppycrawl.tools</groupId>
                            <artifactId>checkstyle</artifactId>
                            <version>10.2</version>
                        </dependency>
                    </dependencies>
                </plugin>
            </plugins>
        </pluginManagement>
    </build>
</profile>
```

will print message multiple times `[Fatal Error] control.xml:1:1: Content is not allowed in prolog.`

```
[INFO] --- maven-checkstyle-plugin:3.1.2:check (default-cli) @ my.project ---
[Fatal Error] control.xml:1:1: Content is not allowed in prolog.
[Fatal Error] control.xml:1:1: Content is not allowed in prolog.
[Fatal Error] control.xml:1:1: Content is not allowed in prolog.
[Fatal Error] control.xml:1:1: Content is not allowed in prolog.
[INFO] You have 0 Checkstyle violations.
```

using the version `10.1` as in

```xml
<profile>
    <id>checkstyle</id>

    <build>
        <pluginManagement>
            <plugins>
                <plugin>
                    <groupId>org.apache.maven.plugins</groupId>
                    <artifactId>maven-checkstyle-plugin</artifactId>
                    <version>3.1.2</version>

                    <configuration>
                        <configLocation>.checkstyle/config.xml</configLocation>
                        <includeTestSourceDirectory>true</includeTestSourceDirectory>
                    </configuration>

                    <dependencies>
                        <dependency>
                            <groupId>com.puppycrawl.tools</groupId>
                            <artifactId>checkstyle</artifactId>
                            <version>10.1</version>
                        </dependency>
                    </dependencies>
                </plugin>
            </plugins>
        </pluginManagement>
    </build>
</profile>
```

will only print 

```
[INFO] --- maven-checkstyle-plugin:3.1.2:check (default-cli) @ my.project ---
[INFO] You have 0 Checkstyle violations.
```

I tried executing `mvn` with `--debug`, but no sensible output is printed:

with **10.1**:

```
[INFO] --- maven-checkstyle-plugin:3.1.2:check (default-cli) @ my.project ---
[DEBUG] Configuring mojo org.apache.maven.plugins:maven-checkstyle-plugin:3.1.2:check from plugin realm ClassRealm[plugin>org.apache.maven.plugins:maven-checkstyle-plugin:3.1.2, parent: jdk.internal.loader.ClassLoaders$AppClassLoader@46fbb2c1]
[DEBUG] Configuring mojo 'org.apache.maven.plugins:maven-checkstyle-plugin:3.1.2:check' with basic configurator -->
[DEBUG]   (f) cacheFile = /home/jenkins/workspace/checkstyle-fatals/my.project/target/checkstyle-cachefile
[DEBUG]   (f) checkstyleRulesHeader = <?xml version="1.0"?>
<!DOCTYPE module PUBLIC "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">

[DEBUG]   (f) configLocation = .checkstyle/config.xml
[DEBUG]   (f) consoleOutput = false
[DEBUG]   (f) encoding = UTF-8
[DEBUG]   (f) failOnViolation = true
[DEBUG]   (f) failsOnError = false
[DEBUG]   (f) headerLocation = LICENSE.txt
[DEBUG]   (f) includeResources = true
[DEBUG]   (f) includeTestResources = true
[DEBUG]   (f) includeTestSourceDirectory = true
[DEBUG]   (f) includes = **\/*.java
[DEBUG]   (f) logViolationCountToConsole = true
[DEBUG]   (f) logViolationsToConsole = true
[DEBUG]   (f) maxAllowedViolations = 0
[DEBUG]   (f) omitIgnoredModules = false
[DEBUG]   (f) outputFile = /home/jenkins/workspace/checkstyle-fatals/my.project/target/checkstyle-result.xml
[DEBUG]   (f) outputFileFormat = xml
[DEBUG]   (f) plugin = Component Descriptor: role: 'org.apache.maven.plugin.Mojo', implementation: 'org.apache.maven.plugins.checkstyle.CheckstyleViolationCheckMojo', role hint: 'org.apache.maven.plugins:maven-checkstyle-plugin:3.1.2:check'
role: 'org.apache.maven.plugin.Mojo', implementation: 'org.apache.maven.plugins.checkstyle.CheckstyleReport', role hint: 'org.apache.maven.plugins:maven-checkstyle-plugin:3.1.2:checkstyle'
role: 'org.apache.maven.plugin.Mojo', implementation: 'org.apache.maven.plugins.checkstyle.CheckstyleAggregateReport', role hint: 'org.apache.maven.plugins:maven-checkstyle-plugin:3.1.2:checkstyle-aggregate'
role: 'org.apache.maven.plugin.Mojo', implementation: 'org.apache.maven.plugins.checkstyle.HelpMojo', role hint: 'org.apache.maven.plugins:maven-checkstyle-plugin:3.1.2:help'
---
[DEBUG]   (f) project = MavenProject: example:my.project:1.0.0-SNAPSHOT @ /home/jenkins/workspace/checkstyle-fatals/my.project/pom.xml
[DEBUG]   (f) resourceIncludes = **/*.properties
[DEBUG]   (f) resources = [Resource {targetPath: null, filtering: false, FileSet {directory: /home/jenkins/workspace/checkstyle-fatals/my.project/src/main/resources, PatternSet [includes: {}, excludes: {}]}}]
[DEBUG]   (f) rulesFiles = /home/jenkins/workspace/checkstyle-fatals/my.project/target/checkstyle-rules.xml
[DEBUG]   (f) skip = false
[DEBUG]   (f) skipExec = false
[DEBUG]   (f) suppressionsFileExpression = checkstyle.suppressions.file
[DEBUG]   (f) testResources = [Resource {targetPath: null, filtering: false, FileSet {directory: /home/jenkins/workspace/checkstyle-fatals/my.project/src/test/resources, PatternSet [includes: {}, excludes: {}]}}]
[DEBUG]   (f) violationSeverity = error
[DEBUG] -- end configuration --
[DEBUG] executeCheckstyle start headerLocation : LICENSE.txt
[DEBUG] The resources directory '/home/jenkins/workspace/checkstyle-fatals/my.project/src/main/resources' does not exist or is not a directory.
[DEBUG] The resources directory '/home/jenkins/workspace/checkstyle-fatals/my.project/src/test/resources' does not exist or is not a directory.
[DEBUG] Added 0 files to process.
[DEBUG] request.getConfigLocation() .checkstyle/config.xml
[DEBUG] The resource '.checkstyle/config.xml' was not found with resourceLoader org.codehaus.plexus.resource.loader.ThreadContextClasspathResourceLoader.
[DEBUG] The resource '.checkstyle/config.xml' was found as /home/jenkins/workspace/checkstyle-fatals/.checkstyle/config.xml.
[DEBUG] headerLocation LICENSE.txt
[DEBUG] The resource 'LICENSE.txt' was not found with resourceLoader org.codehaus.plexus.resource.loader.FileResourceLoader.
[DEBUG] JarResourceLoader : trying to load "jar:file:/home/jenkins/workspace/checkstyle-fatals/%3F/.m2/repository/com/puppycrawl/tools/checkstyle/10.1/checkstyle-10.1.jar"
[DEBUG] The resource 'LICENSE.txt' was not found with resourceLoader org.codehaus.plexus.resource.loader.JarResourceLoader.
[DEBUG] URLResourceLoader: No valid URL 'LICENSE.txt'
[DEBUG] URLResourceLoader: No valid URL 'LICENSE.txt'
[DEBUG] The resource 'LICENSE.txt' was not found with resourceLoader org.codehaus.plexus.resource.loader.URLResourceLoader.
[DEBUG] Unable to process header location: LICENSE.txt
[DEBUG] Checkstyle will throw exception if ${checkstyle.header.file} is used
[INFO] You have 0 Checkstyle violations.
```

but with **10.2** checkstyle will print additional fatals in the log:

```
[INFO] --- maven-checkstyle-plugin:3.1.2:check (default-cli) @ my.project ---
[DEBUG] Configuring mojo org.apache.maven.plugins:maven-checkstyle-plugin:3.1.2:check from plugin realm ClassRealm[plugin>org.apache.maven.plugins:maven-checkstyle-plugin:3.1.2, parent: jdk.internal.loader.ClassLoaders$AppClassLoader@46fbb2c1]
[DEBUG] Configuring mojo 'org.apache.maven.plugins:maven-checkstyle-plugin:3.1.2:check' with basic configurator -->
[DEBUG]   (f) cacheFile = /home/jenkins/workspace/checkstyle-fatals/my.project/target/checkstyle-cachefile
[DEBUG]   (f) checkstyleRulesHeader = <?xml version="1.0"?>
<!DOCTYPE module PUBLIC "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">

[DEBUG]   (f) configLocation = .checkstyle/config.xml
[DEBUG]   (f) consoleOutput = false
[DEBUG]   (f) encoding = UTF-8
[DEBUG]   (f) failOnViolation = true
[DEBUG]   (f) failsOnError = false
[DEBUG]   (f) headerLocation = LICENSE.txt
[DEBUG]   (f) includeResources = true
[DEBUG]   (f) includeTestResources = true
[DEBUG]   (f) includeTestSourceDirectory = true
[DEBUG]   (f) includes = **\/*.java
[DEBUG]   (f) logViolationCountToConsole = true
[DEBUG]   (f) logViolationsToConsole = true
[DEBUG]   (f) maxAllowedViolations = 0
[DEBUG]   (f) omitIgnoredModules = false
[DEBUG]   (f) outputFile = /home/jenkins/workspace/checkstyle-fatals/my.project/target/checkstyle-result.xml
[DEBUG]   (f) outputFileFormat = xml
[DEBUG]   (f) plugin = Component Descriptor: role: 'org.apache.maven.plugin.Mojo', implementation: 'org.apache.maven.plugins.checkstyle.CheckstyleViolationCheckMojo', role hint: 'org.apache.maven.plugins:maven-checkstyle-plugin:3.1.2:check'
role: 'org.apache.maven.plugin.Mojo', implementation: 'org.apache.maven.plugins.checkstyle.CheckstyleReport', role hint: 'org.apache.maven.plugins:maven-checkstyle-plugin:3.1.2:checkstyle'
role: 'org.apache.maven.plugin.Mojo', implementation: 'org.apache.maven.plugins.checkstyle.CheckstyleAggregateReport', role hint: 'org.apache.maven.plugins:maven-checkstyle-plugin:3.1.2:checkstyle-aggregate'
role: 'org.apache.maven.plugin.Mojo', implementation: 'org.apache.maven.plugins.checkstyle.HelpMojo', role hint: 'org.apache.maven.plugins:maven-checkstyle-plugin:3.1.2:help'
---
[DEBUG]   (f) project = MavenProject: example:my.project:1.0.0-SNAPSHOT @ /home/jenkins/workspace/checkstyle-fatals/my.project/pom.xml
[DEBUG]   (f) resourceIncludes = **/*.properties
[DEBUG]   (f) resources = [Resource {targetPath: null, filtering: false, FileSet {directory: /home/jenkins/workspace/checkstyle-fatals/my.project/src/main/resources, PatternSet [includes: {}, excludes: {}]}}]
[DEBUG]   (f) rulesFiles = /home/jenkins/workspace/checkstyle-fatals/my.project/target/checkstyle-rules.xml
[DEBUG]   (f) skip = false
[DEBUG]   (f) skipExec = false
[DEBUG]   (f) suppressionsFileExpression = checkstyle.suppressions.file
[DEBUG]   (f) testResources = [Resource {targetPath: null, filtering: false, FileSet {directory: /home/jenkins/workspace/checkstyle-fatals/my.project/src/test/resources, PatternSet [includes: {}, excludes: {}]}}]
[DEBUG]   (f) violationSeverity = error
[DEBUG] -- end configuration --
[DEBUG] executeCheckstyle start headerLocation : LICENSE.txt
[DEBUG] The resources directory '/home/jenkins/workspace/checkstyle-fatals/my.project/src/main/resources' does not exist or is not a directory.
[DEBUG] The resources directory '/home/jenkins/workspace/checkstyle-fatals/my.project/src/test/resources' does not exist or is not a directory.
[DEBUG] Added 0 files to process.
[DEBUG] request.getConfigLocation() .checkstyle/config.xml
[DEBUG] The resource '.checkstyle/config.xml' was not found with resourceLoader org.codehaus.plexus.resource.loader.JarResourceLoader.
[DEBUG] URLResourceLoader: No valid URL '.checkstyle/config.xml'
[DEBUG] URLResourceLoader: No valid URL '.checkstyle/config.xml'
[DEBUG] The resource '.checkstyle/config.xml' was not found with resourceLoader org.codehaus.plexus.resource.loader.URLResourceLoader.
[DEBUG] The resource '.checkstyle/config.xml' was found as /home/jenkins/workspace/checkstyle-fatals/.checkstyle/config.xml.
[DEBUG] headerLocation LICENSE.txt
[DEBUG] JarResourceLoader : trying to load "jar:file:/home/jenkins/workspace/checkstyle-fatals/%3F/.m2/repository/com/puppycrawl/tools/checkstyle/10.2/checkstyle-10.2.jar"
[DEBUG] The resource 'LICENSE.txt' was not found with resourceLoader org.codehaus.plexus.resource.loader.JarResourceLoader.
[DEBUG] URLResourceLoader: No valid URL 'LICENSE.txt'
[DEBUG] URLResourceLoader: No valid URL 'LICENSE.txt'
[DEBUG] The resource 'LICENSE.txt' was not found with resourceLoader org.codehaus.plexus.resource.loader.URLResourceLoader.
[DEBUG] The resource 'LICENSE.txt' was not found with resourceLoader org.codehaus.plexus.resource.loader.FileResourceLoader.
[DEBUG] Unable to process header location: LICENSE.txt
[DEBUG] Checkstyle will throw exception if ${checkstyle.header.file} is used
[Fatal Error] control.xml:1:1: Content is not allowed in prolog.
[Fatal Error] control.xml:1:1: Content is not allowed in prolog.
[INFO] You have 0 Checkstyle violations.
```

Upgrading to 10.3 resulted in the **same** output.
I am running Jenkins on Ubuntu 20.04 and the message will **not** be printed on Windows, so it might be some XML encodings? A search within this repository did not result in any meaningful answers.

It does **not** affect the build result, because checkstyle will not find any violations. Though it triggers for some reason on basically empty projects

```bash
ls ./my.project
pom.xml
```

I also checked the commits between 10.1 and 10.2, but on the first glimpse did not find a specific offender

https://github.com/checkstyle/checkstyle/compare/checkstyle-10.1...checkstyle-10.2

**Comment by romani** [comment]
in this repo we accept issues that use checkstyle CLI to reproduce issue.

it might be smth related to https://github.com/checkstyle/checkstyle/issues/11507but I am not sure

please open issue on maven plugin issue tracker:
https://maven.apache.org/plugins/maven-checkstyle-plugin/issue-management.html
https://issues.apache.org/jira/projects/MCHECKSTYLE/issues/MCHECKSTYLE-408?filter=allopenissues

**Comment by commodis** [comment]
I opened here, because I did not update the maven-plugin and just the checkstyle dependency, which regressed on a minor-version. But thanks for the pointer to the plugin issue tracker! Might be some weird interaction 🤔

**Comment by commodis** [comment]
@rnveach 

I try to set up some way to reproduce it locally, but running the same command with the same repository on windows does not print that message. I ran it on Docker and it did not print the message. I will try installing everything necessary on WSL and check if that is reproducible.

Regarding the concerns:

Is it not possible to use `<configLocation>.checkstyle/config.xml</configLocation>`? I am pretty sure it is picked up just fine and should not matter if I use `${project.basedir}/.checkstyle/config.xml` or not?

I also did not regard them as necessary because they were invoked by `JarResourceLoader` and `URLResourceLoader`, which I assume are for cases when the config is packaged in a JAR and is stored remotely (which does not match my situation).

**Comment by rnveach** [comment]
> Can you point by link where we missed update?

**Comment by rnveach** [comment]
> Can you point by link where we missed update?

Can you point by link where we missed update? But again, we can't upgrade until sevntu does, which is waiting on eclipse-cs.

> Is it not possible to use <configLocation>.checkstyle/config.xml</configLocation>?

I was just stating that this is a change in the log. It seems like something else is going on besides the version upgrade. The file location/naming should not be a problem.

**Comment by romani** [comment]
latest maven plugin and latest checkstyle:
https://github.com/sevntu-checkstyle/checkstyle-samples/blob/e2d5a4fdddf2bf74deb110c8cc0e69a761eb0f05/maven-project/pom.xml#L16

works ok:
https://app.travis-ci.com/github/sevntu-checkstyle/checkstyle-samples/builds/252168291#L357

**Comment by commodis** [comment]
@romani I can try downgrading them in the plugin. `Saxon-HE` looks suspicious to me, since it is a major version jump and is a processor for XML.
